common/src/main/java/io/netty/util/concurrent/AutoScalingEventExecutorChooserFactory.java

```diff
@@ -45,6 +45,32 @@
  */
 public final class AutoScalingEventExecutorChooserFactory implements EventExecutorChooserFactory {
 
+    /**
+     * A container for the utilization metric of a single EventExecutor.
+     * This object is intended to be created once and have its {@code utilization}
+     * field updated periodically.
+     */
+    public static final class AutoScalingUtilizationMetric {
+        final EventExecutor executor;
+        private volatile double utilization;
+
+        AutoScalingUtilizationMetric(EventExecutor executor) {
+            this.executor = executor;
+        }
+
+        /**
+         * Returns the most recently calculated utilization for the associated executor.
+         * @return a value from 0.0 to 1.0.
+         */
+        public double utilization() {
+            return utilization;
+        }
+
+        void setUtilization(double utilization) {
+            this.utilization = utilization;
```

Comment:
nit: I think you could use lazySet for this one as it will only ever be modified from the same thread.